linux/net/rina/pdu.c

```diff
@@ -287,3 +288,9 @@ int pdu_destroy(struct pdu * p)
         return 0;
 }
 EXPORT_SYMBOL(pdu_destroy);
+
+uint_t pdu_length(struct pdu * pdu) {
+	return buffer_length(pdu->buffer) +
+		serdes_pci_size(pci_type(pdu->pci));
+}
+EXPORT_SYMBOL(pdu_length);
```

Comment:
Fixed that part after lot of invocations to god (thanks Vincenzo). 

I added a new issue in IRATI stack for the dt constant structure missing fields for rate based pci fields sizes. Now parse_dt_cons in rnl_utils has the following fix, which can be replaced once the support for this field is introduced.

```
/* FIXME: To modify once this field are supported by rnl. */
dt_cons->rate = 4;
dt_cons->frame = 4;
```